src/Metadata/Property/Factory/ExtractorPropertyMetadataFactory.php

```diff
@@ -48,10 +51,8 @@ public function create(string $resourceClass, string $property, array $options =
             }
         }
 
-        $isInterface = interface_exists($resourceClass);
-
         if (
-            !property_exists($resourceClass, $property) && !$isInterface ||
+            $this->doesIdentifierExist($resourceClass, $property) ||
```

Comment:
Can you use `$this->extractor->isReadable` instead?

Reply:
Ah.  Possible.  I'll need to dig back into this, as it's obviously been a while.  I'll get back too you.

Sorry for the delay.